### DISCUSSION TITLE
ParallelGPT Template is not returning a value

### DISCUSSION THREAD

**Discussion by broderius** [AntlerVC/firetable#1277] [Support / Q&A] [answered]
I went through the video tutorial to set up but its not returning a value.

Here is the logs from Google Cloud: 
![image](https://github.com/rowyio/rowy/assets/30006479/1c39e51a-fa5e-49e8-a246-a6af7287f9b9)

What should I do to fix it?

**Comment by harinij** [comment, marked as answer]
Hi @broderius from the error message it appears that the `openai` secret is not set in your Google Cloud Project. You can follow these steps to add the secret and that should resolve the issue. Or double check if your secret is named something else.

- Go to secret manager for your project from [here](https://console.cloud.google.com/projectselector2/security/secret-manager?referrer=search&project=_&supportedpurview=project)
- Then click `create secret` 
<img width="320" alt="Screenshot 2023-06-06 at 2 54 36 pm" src="https://github.com/rowyio/rowy/assets/307298/a015ba86-deaa-41dd-b75d-652c6886bf6d">  

- Give the secret a name `openai` and add the API key from your OpenAI account
<img width="500" alt="Screenshot 2023-06-06 at 2 54 29 pm" src="https://github.com/rowyio/rowy/assets/307298/3817e777-72ff-4c8e-be2c-9c04f5d1ca42">
